### PR TITLE
igraph: update to version 0.8.0

### PR DESCRIPTION
**Pull request by szhorvat**
#### Description

igraph: update to version 0.8.0

 * update to version 0.8.0
 * add external_glpk variant and make it default
 * change to github PortGroup
 * remove indirect dependencies
 * adapt livecheck
 * add science category
 * update description

py-igraph: update to version 0.8.0

 * update to version 0.8.0
 * remove dependency on port:igraph, as python-igraph now includes its embedded copy
 * add new dependencies required by embedded igraph
 * build with external glpk (new depends on port:glpk)
 * add python.versions 37 and 38
 * remove no longer needed patches
 * add science category
 * update description

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.6 18G3020
Xcode 11.3.1 11C504

###### Verification 

Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?

### CODE REVIEW

**Comment by szhorvat** [comment]
@nerdling I am not experienced with portfiles, and the following issues should be checked:

 - The default `livecheck` with the `github` PortGroup does not appear to just check for a new release, therefore it already claims that this is outdated. How do I make it simply check for new releases?

 - `py-igraph` and related ports are currently at an old version, and they do not compile with 0.8.0 of the igraph C library. A new version of igraph's Python interface (with full Python 3 support) is planned to be released relatively soon (~weeks). However, we would like to have the C library updated to 0.8.0 in MacPorts right away, before the new version of `python-igraph` is ready. Can the `py-igraph` ports simply be marked as requiring `igraph` 0.7.1, while still having 0.8.0 available in MacPorts?

**Comment by szhorvat** [comment]
Actually, it looks like a new python-igraph release may happen relatively soon. I might make sense to wait for that and update everything together.

**Comment by szhorvat** [comment]
I just added an `external_glpk` variant, to use MacPorts's GLPK instead of igraph's embedded (and outdated) GLPK. This provides a performance boost for the functions `igraph_feedback_arc_set` and `igraph_community_optimal_modularity`.

**Comment by pmetzger** [comment]
@nerdling do you have an opinion here? Otherwise someone should probably commit this.

**Comment by szhorvat** [comment]
@pmetzger Thanks for following up. This would break the `py-igraph` ports if committed now. Let us wait for a week or two, by which time there should be a new python-igraph release. Then both the igraph and the py-igraph ports can be safely updated simultaneously.

**Comment by nerdling** [comment]
> It may not be a bad idea to make this the default.

In the mean time, did you want to make that variant the default?

**Comment by pmetzger** [comment]
> This would break the py-igraph ports if committed now. Let us wait for a week or two, by which time there should be a new python-igraph release. Then both the igraph and the py-igraph ports can be safely updated simultaneously.

Okay, so we should hold off for the moment?

**Comment by szhorvat** [comment]
> In the mean time, did you want to make that variant the default?

I was waiting for confirmation. I made it the default now.

> Okay, so we should hold off for the moment?

Yes, if all goes well, it'll take only a week (much less than I expected when opening this PR).

**Comment by pmetzger** [comment]
Please let us know when to do the merge, then.

**Comment by szhorvat** [comment]
@nerdling This should be good to go now, but please take a careful look, as I'm not experienced with portfiles. I updated py-igraph and removed the patches, as they no longer seem to be necessary (now python-igraph includes its private igraph C library).

**Comment by szhorvat** [comment]
I'm not sure why those failed ... can someone help me and link to where I can see the port build log from those tests? It built fine on my machine.

**Comment by mf2k** [comment]
error is:

```
Finding out version number/string... fatal: not a git repository: /opt/local/var/macports/build/_Users_travis_build_macports_macports-ports_python_py-igraph/py37-igraph/work/python-igraph-0.8.0/vendor/source/igraph/../../../.git/modules/vendor/source/igraph
0.8.0-pre+
libtoolize or glibtoolize not found or not in the path!
We are going to build the C core of igraph.
  Source folder: vendor/source/igraph
  Build folder: vendor/build/igraph
  Install folder: vendor/install/igraph
 
Bootstrapping igraph...
Could not compile the C core of igraph.

```

**Comment by szhorvat** [comment]
@mf2k Thank you! I saw that after macportsbot posted the links to the logs. `libtool` had to be added as a build dependency. I believe it's fixed now, but let's wait for the travis test to finish.

**Comment by szhorvat** [comment]
The Travis build timed out, but during the time it ran, there was no indication of failure. The other tests succeeded. How do we proceed? Can this be merged now?

File: python/py-igraph/Portfile [hunk 1]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
 PortGroup           python 1.0
 
 name                py-igraph
```
Comment (reneeotten) [inline python/py-igraph/Portfile]: under the ```name``` field, you could add ```python.rootname     python-igraph```. If you do that, the python PG will populate some of the fields automatically and you can simplify the Portfile (i.e., remove some lines) - see comments below.

File: python/py-igraph/Portfile [hunk 2]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
...
+description         Python interface to the igraph library.
 
-long_description    Python interface to the igraph high performance \
-                    graph library
+long_description    Python interface to the igraph library for network analysis \
+                    and graph theory. To enable visualization functionality, \
+                    it is recommended to also install py-cairocffi.
 
 homepage            https://pypi.python.org/pypi/python-igraph/
 master_sites        pypi:p/python-igraph/
```
Comment (reneeotten) [inline python/py-igraph/Portfile]: this line can be removed if you add ```python.rootname```

File: python/py-igraph/Portfile [hunk 3]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
...
 
 distname            python-igraph-${version}
```
Comment (reneeotten) [inline python/py-igraph/Portfile]: this line can be removed if you add ```python.rootname```

File: python/py-igraph/Portfile [hunk 4]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
...
 dist_subdir         python/${distname}/1/
```
Comment (reneeotten) [inline python/py-igraph/Portfile]: this shouldn't be necessary... or?

File: python/py-igraph/Portfile [hunk 5]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
...
+                        port:libtool port:autoconf port:automake port:bison port:flex
 
-    post-patch {
-        reinplace s|@PREFIX@|${prefix}|g ${worksrcpath}/setup.cfg
-        reinplace s|@PREFIX@|${prefix}|g ${worksrcpath}/setup.py
-    }
+    build.env-append    IGRAPH_EXTRA_CONFIGURE_ARGS=--with-external-glpk
 
     livecheck.type      none
 } else {
```
Comment (reneeotten) [inline python/py-igraph/Portfile]: the ```else```-block can go if you set ```python.rootname```

File: python/py-igraph/Portfile [hunk 6]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
...
-checksums           rmd160  0d843b17da829ee531336e9bbc5dbf064a270a5e \
-                    sha256  f155bca04d5e500cb3cc087d25ac9e09077959392d245b3cab588bd9b541f0d0
+checksums           rmd160  da5dbc21c429988ab9ee73c084b2e8c0fa2dcae2 \
+                    sha256  47e6bb48ec7dbfddbd89cf064a24b271783a1490fc688ebce17fbd652bcdab8e \
+                    size    3805258                    
 
 if {${name} ne ${subport}} {
-    depends_build       port:pkgconfig port:py${python.version}-setuptools
-    depends_lib-append  port:igraph
+    depends_lib         port:py${python.version}-texttable \
```
Comment (reneeotten) [inline python/py-igraph/Portfile]: please use ```depends_lib-append``` instead, as not to overwrite the dependencies added by the PG

File: python/py-igraph/Portfile [hunk 7]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
...
+                        port:gmp port:libxml2 port:glpk
 
-    patchfiles          patch-setup.cfg.diff \
-                        patch-setup.py.diff
+    depends_build       port:py${python.version}-setuptools \
```
Comment (reneeotten) [inline python/py-igraph/Portfile]: please use ```depends_build-append``` instead, as not to overwrite the dependencies added by the PG

File: python/py-igraph/Portfile [hunk 8]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
...
+                        port:libtool port:autoconf port:automake port:bison port:flex
```
Comment (reneeotten) [inline python/py-igraph/Portfile]: I would personally prefer to list those dependencies all on separate lines

File: python/py-igraph/Portfile (hunk 4 above)
Comment (szhorvat) [reply python/py-igraph/Portfile]: I don't know. It was like this. @nerdling ?

**Comment by szhorvat** [comment]
@reneeotten 

Thanks for the comments! Can you check again please? Once it's fine, I'll squash the commits.

> It's probably still preferred to patch `setup.py` and make sure that the only libraries looked for are those installed by MacPorts in ${prefix}.

The part of `setup.py` that the patch still applied to isn't used for this port because the source of the C igraph library is now included in python-igraph. Even if python-igraph were configured to use a separate igraph C library, it would use pkgconfig to find it. If it needed to go to the fallback, that would indicate a bug in the portfile, which is best detected early (better to let it fail than pick up C/igraph from MacPorts the incorrect way).

Do you still want me to apply the patch?

File: python/py-igraph/Portfile (hunk 4 above)
Comment (reneeotten) [reply python/py-igraph/Portfile]: this is likely a left-over from a previous [stealth update](https://trac.macports.org/wiki/PortfileRecipes#stealth-updates). It builds fine without this line and should be removed.

Comment (szhorvat) [reply python/py-igraph/Portfile]: It's already removed, so marking this as resolved.

**Comment by szhorvat** [comment]
I went ahead and squashed the commits. Is it good to go now or do you still want me to re-add that patch? (Note: The code the patch affects does not and *should not* run when compiling the port.)

**Comment by szhorvat** [comment]
@nerdling Thanks! If there is any issue with `igraph` or `py-igraph` in the future, feel free to ask me.

**Pull request merged**